api/products.thrift

```diff
@@ -0,0 +1,78 @@
+// -------------------------------------------------------------------------
+//                     The CodeChecker Infrastructure
+//   This file is distributed under the University of Illinois Open Source
+//   License. See LICENSE.TXT for details.
+// -------------------------------------------------------------------------
+
+include "shared.thrift"
+
+namespace py ProductManagement
+namespace js ProductManagement
+
+
+/*
+struct PrivilegeRecord {
+  1: string   name,
+  2: bool     isGroup
+}
+typedef list<PrivilegeRecord> PrivilegeRecords
+*/
+
+struct DatabaseConnection {
+  1:          string engine,         // The database engine, such as "sqlite" or "postgresql".
+  2:          string host,
+  3:          i32    port,
+  4:          string username_b64,
+  5: optional string password_b64,   // Database password is NOT sent server->client!
+  6:          string database
+}
+
+/* ProductConfiguration carries administrative data regarding product settings. */
+struct ProductConfiguration {
+  1:          i64                id,
+  2:          string             endpoint,
+  3:          string             displayedName_b64,
+  4:          string             description_b64,
+  5: optional DatabaseConnection connection
+}
+typedef list<ProductConfiguration> ProductConfigurations
+
+/* Product carries data to the end user's product list and tasks. */
+struct Product {
+  1: i64    id,
+  2: string endpoint,
+  3: string displayedName_b64,
+  4: string description_b64,
+  5: bool   connected,         // Indicates that the server could set up the database connection properly.
+  6: bool   accessible         // Indicates whether the current user can access this product.
+}
+typedef list<Product> Products
+
+service productService {
```

Comment:
The other service names start with the prefix `codeChecker`:
- codeCheckerAuthentication
- codeCheckerDBAccess

I think we should use the same convention here (E.g.: codeCheckerProduct)